control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -1339,11 +1335,12 @@ func (r *HostedControlPlaneReconciler) reconcile(ctx context.Context, hostedCont
 
 	if !r.IsCPOV2 {
 		// Reconcile image registry operator
-		r.Log.Info("Reconciling Image Registry Operator")
-		if err := r.reconcileImageRegistryOperator(ctx, hostedControlPlane, releaseImageProvider, userReleaseImageProvider, createOrUpdate); err != nil {
-			return fmt.Errorf("failed to reconcile image registry operator: %w", err)
+		if capabilities.IsImageRegistryCapabilityEnabled(hostedControlPlane.Spec.Capabilities) {
```

Comment:
you would need to create a predicate to stop reconciliation in v2 here: https://github.com/openshift/hypershift/blob/main/control-plane-operator/controllers/hostedcontrolplane/v2/registryoperator/component.go#L34

see https://github.com/openshift/hypershift/blob/main/control-plane-operator/controllers/hostedcontrolplane/v2/csi/kubevirt/component.go#L48 as example.

Reply:
Done.